### PR TITLE
Speedup mask filter.

### PR DESCRIPTION
**Pull request by FloWuenne**
## PR checklist

This PR replaces the mask filtering code by more efficient code that speeds up the function, particularly for larger datasets.

- [X] This comment contains a description of changes (with reason).
- [x] Make sure your code lints (`nf-core lint`).
- [x] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [x] `CHANGELOG.md` is updated.

### CODE REVIEW

**Comment by FloWuenne** [comment]
This PR will resolve #38 !

File: bin/maskfilter.py [hunk 1]
```diff
@@ -55,28 +55,37 @@ def get_args():
...
+    elif min_area:
+        valid_labels = np.array([label for label, area in label_area_map.items() if area >= min_area])
+        retained_masks = np.isin(labeled_mask, valid_labels) * labeled_mask
+        large_labels = 0
+        small_labels = len(labeled_mask) - len(valid_labels)
+        relabeled_mask = label(retained_masks, background=0)
+    elif max_area:
+        valid_labels = np.array([label for label, area in label_area_map.items() if area <= max_area])
+        retained_masks = np.isin(labeled_mask, valid_labels) * labeled_mask
+        large_labels = len(labeled_mask) - len(valid_labels)
```
Comment (kbestak) [inline bin/maskfilter.py:81]:
```suggestion
        large_labels = len(labeled_mask) - len(valid_labels)
        small_labels = 0
```

**Review by kbestak — COMMENTED** [review]
If only max_area is provided, small_labels is not defined.

**Review by kbestak — APPROVED** [review]
LGTM!

**Pull request merged**